packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -205,6 +218,9 @@ export class EcsOptimizedImage implements ec2.IMachineImage {
    * @param hardwareType ECS-optimized AMI variant to use
    */
   public static amazonLinux2023(hardwareType = AmiHardwareType.STANDARD, options: EcsOptimizedImageOptions = {}): EcsOptimizedImage {
+    if (options.kernel) {
+      throw new Error('Kernel version can only be specified for Amazon Linux 2');
+    }
```

Comment:
I think we should provide the ability to specify the `kernel` property for all Linux generations.
What about declaring dedicated interfaces?
```ts
export interface EcsOptimizedAmazonLinux2023ImageOptions extends EcsOptimizedImageOptions {
  /**
   * The kernel version to use with Amazon Linux 2023
   *
   * @default - The recommended kernel version
   */
  readonly kernel?: ec2.AmazonLinux2022Kernel;
}
...
```

Reply:
I considered that but decided against it for 2 reasons:
1. The [AMI metadata SSM parameters](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/retrieve-ecs-optimized_AMI.html) do not seem to contain explicit kernel path except for Amazon Linux 2 AMIs

2. introducing a new interface like `EcsOptimizedAmazonLinux2023ImageOptions` seems to cause lint errors like:
```
Some packages seem to have undergone breaking API changes. Please avoid.
```

Please let me know if I am misunderstanding something. 🙏

Reply:
Thanks for the follow-up, you're right about point 1 👍

However, I think that the idea of adding an interface for Amazon Linux 2 `kernel` property is still valuable, given that we would be able to simplify the implementation and remove the manual validations:
```ts
export interface EcsOptimizedAmazonLinux2ImageOptions extends EcsOptimizedImageOptions {
  /**
   * The kernel version to use with Amazon Linux 2
   *
   * @default - The recommended kernel version
   */
  readonly kernel?: ec2.AmazonLinux2Kernel;
}
```
We would then need to change options type for the `amazonLinux2` method:
```ts
public static amazonLinux2(hardwareType = AmiHardwareType.STANDARD, options: EcsOptimizedAmazonLinux2ImageOptions = {}): EcsOptimizedImage { ... }
```
What do you think?

On point 2, you can allow the change by adding an entry to `allowed-breaking-changes.txt` ([example](https://github.com/aws/aws-cdk/blob/b0490640a25eb1946aa9db748cf85867e841f243/allowed-breaking-changes.txt#L92)).

Reply:
@Konosh93 

Thanks for the follow-up, you're right about point 1 👍

However, I think that the idea of adding an interface for Amazon Linux 2 `kernel` property is still valuable, given that we would be able to simplify the implementation and remove the manual validations:
```ts
export interface EcsOptimizedAmazonLinux2ImageOptions extends EcsOptimizedImageOptions {
  /**
   * The kernel version to use with Amazon Linux 2
   *
   * @default - The recommended kernel version
   */
  readonly kernel?: ec2.AmazonLinux2Kernel;
}
```
We would then need to change options type for the `amazonLinux2` method:
```ts
public static amazonLinux2(hardwareType = AmiHardwareType.STANDARD, options: EcsOptimizedAmazonLinux2ImageOptions = {}): EcsOptimizedImage { ... }
```
What do you think?

On point 2, you can allow the change by adding an entry to `allowed-breaking-changes.txt` ([example](https://github.com/aws/aws-cdk/blob/b0490640a25eb1946aa9db748cf85867e841f243/allowed-breaking-changes.txt#L92)).

Reply:
@lpizzinidev 

Thanks for the tips and detailed code examples 🙏 

> adding an interface for Amazon Linux 2 kernel property is still valuable

Indeed, I added proposed interface and and updated the breaking change file accordingly.